libs/chain-signatures/contract/src/lib.rs

```diff
@@ -810,206 +579,133 @@ impl VersionedMpcContract {
     #[init(ignore_state)]
     #[handle_result]
     pub fn migrate() -> Result<Self, Error> {
-        let old: VersionedMpcContract =
-            env::state_read().ok_or(InvalidState::ContractStateIsMissing)?;
-        match old {
-            VersionedMpcContract::V0(mpc_contract_v0) => {
-                Ok(VersionedMpcContract::V1(MpcContractV1 {
-                    config: ConfigV1::default(),
-                    protocol_state: mpc_contract_v0.protocol_state,
-                    pending_requests: mpc_contract_v0.pending_requests,
-                    request_by_block_height: Vector::new(StorageKey::RequestsByTimestamp),
-                    proposed_updates: ProposedUpdates::default(),
-                }))
+        if let Some(legacy_contract::VersionedMpcContract::V1(state)) =
+            env::state_read::<legacy_contract::VersionedMpcContract>()
+        {
+            // migrate config
+            let mut config = Config::default();
+            config.request_timeout_blocks = state.config.request_timeout_blocks;
+            config.max_num_requests_to_remove = state.config.max_num_requests_to_remove;
+            // migrate signature requests:
+            let mut request_by_block_height: Vector<(BlockHeight, SignatureRequest)> =
+                Vector::new(StorageKey::RequestsByTimestamp);
+            let mut pending_requests: LookupMap<SignatureRequest, YieldIndex> =
+                LookupMap::new(StorageKey::PendingRequests);
```

Comment:
Do these not just get automatically inherited due to using the same storage key as the legacy contract?

Reply:
I was also wondering about that.

Reply:
In V2 we use `near_sdk::store::LookupMap`, in V0 and V1 we used `near_sdk::collections::LookupMap`.
Looking at the code in [near-sdk](https://github.com/near/near-sdk-rs/blob/546ff5efe9197beff372a28fe45b7c201a90e244/near-sdk/src/store/vec/mod.rs#L195), it seems that `Vector::new(storage_key)` sets the length to 0, so I guess I would need a different mechanism for loading that vec?
I would suggest to tackle this in a separate PR, because the signature state needs some other improvements as well (c.f. https://github.com/Near-One/mpc/issues/241)